examples/tests/valid/inflight_class_inside_inflight_closure.w

```diff
@@ -30,3 +30,14 @@ let f = p.preflight_method();
 test "it works" {
   f.invoke("text");
 }
+
+test "inflight class inside closure captures from closure" {
```

Comment:
Very nice catch.

For outer inflight classes, the compiler detects the attempt to capture mutable/reassignable and emits an error.
For inner inflight classes, the compiler allows the capture and only the mutability works (but reassignablity doesn't because the inflight class is defined in a separate file).

I'll add an issue to track this (I think we shouold be able to merge all the capture visitors together and then we can run these checks there).

Reply:
https://github.com/winglang/wing/issues/2729